src/map.cpp

```diff
@@ -137,4 +138,93 @@ Map::~Map()
     close(mapfd_);
 }
 
+void MapManager::Add(std::unique_ptr<IMap> map)
+{
+  auto name = map->name_;
+  if (name.empty())
+    throw std::runtime_error("Map without name cannot be stored");
+
+  auto count = maps_by_name_.count(name);
+  if (count > 0)
+    throw std::runtime_error("Map with name: @" + name + " already exists");
+
+  std::shared_ptr<IMap> smap = std::move(map);
+  auto id = maps_by_id_.size();
+
+  maps_by_name_[name] = smap;
+  maps_by_id_.emplace_back(smap);
+
+  smap->id = id;
+}
+
+bool MapManager::Has(const std::string &name)
+{
+  return maps_by_name_.find(name) != maps_by_name_.end();
+}
+
+IMap &MapManager::Lookup(const std::string &name)
+{
+  auto search = maps_by_name_.find(name);
+  if (search == maps_by_name_.end())
+  {
+    throw std::runtime_error("No map with name @" + name + " defined");
```

Comment:
Might make more sense to return an `std::optional` instead of throwing exceptions so callers can decide how to handle errors. A missing map isn't really an exceptional situation. Same for other lookup functions below.

Reply:
Yeah sounds good